HLTrigger/HLTcore/src/HLTPrescaleProvider.cc

```diff
@@ -363,9 +363,9 @@ void HLTPrescaleProvider::checkL1TGlobalUtil() const {
 }
 
 template <>
-unsigned int HLTPrescaleProvider::prescaleValue<unsigned int>(const edm::Event& iEvent,
-                                                              const edm::EventSetup& iSetup,
-                                                              const std::string& trigger) {
+float HLTPrescaleProvider::prescaleValue<float>(const edm::Event& iEvent,
+                                                const edm::EventSetup& iSetup,
+                                                const std::string& trigger) {
   const int set(prescaleSet(iEvent, iSetup));
   return set < 0 ? 1 : hltConfigProvider_.prescaleValue<unsigned int>(static_cast<unsigned int>(set), trigger);
```

Comment:
I guess `prescaleValue<unsigned int>` should be changed to `prescaleValue<float>` ?

Reply:
In line 370, <float> will not work due to the explicit type check: https://github.com/cms-sw/cmssw/blob/master/HLTrigger/HLTcore/interface/HLTConfigProvider.h#L186-L195

Reply:
Ah, so `prescaleValue<double>`... but then all code should probably use `double` instead of `float`, as you pointed out.